selfhost/lexer.jakt

```diff
@@ -302,9 +236,75 @@ struct Lexer implements(ThrowingIterable<Token>) {
     input: [u8]
     compiler: Compiler
     comment_contents: [u8]?
+    illegal_cpp_keywords: [String:bool]
 
     fn lex(compiler: Compiler) throws -> [Token] {
-        mut lexer = Lexer(index: 0, input: compiler.current_file_contents, compiler, comment_contents: None)
+        let illegal_cpp_keywords: [String:bool] = [
```

Comment:
How about a set instead?

Reply:
Sure, performance seems around the same for both, updated it.

Reply:
Yeah, didn't expect a performance difference, just less noise and a more proper data structure.